### PR TITLE
chore(`data-stores`): move `@tanstack/query` based custom hooks to `help-center`

### PR DESCRIPTION
**Pull request by flootr**
## Proposed Changes

The intent for the `@automattic/data-stores` package is to hold `@wordpress/data`-based stores. We therefore move custom query and mutation hooks (based on `@tanstack/query`) to a more fitting place. It's mostly `@automattic/help-center` related hooks which we move to a `data/` directory inside that package.

Please note there are a few remaining hooks which require a bit more thought on where to put them and/or are otherwise being worked on.

## Testing Instructions

Can and should be reviewed commit by commit. The first commit moves any hooks from `support-queries/` in one go where as all other commits move either isolated hooks or hooks which depend on each other in some way.

In general, a smoke test of the Help Center (on simple and atomic sites, in Calypso as well as wp-admin) is a good start as most of the hooks are related. In more detail:

Verify the flows under "Still need help" work as expected. On the "Public Forums" one (while on a site with a free plan) verify the site analysis runs when providing a different site url than the site you're currently on.

Verify there are no type errors and no usages of the moved hooks where imports aren't updated accordingly.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: packages/help-center/src/data/use-support-availability.ts [hunk 1]
```diff
@@ -17,7 +17,7 @@ export function useSupportAvailability< SUPPORT_TYPE extends 'CHAT' | 'OTHER' >(
 	enabled = true
 ) {
 	return useQuery< ResponseType< SUPPORT_TYPE >, typeof Error >( {
-		queryKey: [ supportType === 'OTHER' ? 'otherSupportAvailability' : 'chatSupportAvailability' ],
+		queryKey: [ 'support-availability', supportType ],
```
Comment (flootr) [inline packages/help-center/src/data/use-support-availability.ts:20]: This one's a drive-by which simplifies the query key and fixes an eslint violation of `@tanstack/query/exhaustive-deps`.

File: packages/data-stores/src/queries/use-site-intent.ts [hunk 2]
```diff
@@ -1,18 +0,0 @@
-import { useQuery } from '@tanstack/react-query';
```
Comment (flootr) [inline packages/data-stores/src/queries/use-site-intent.ts]: I couldn't find any usages of this hook hence the removal.

File: packages/help-center/src/components/help-center-contact-form.tsx [hunk 3]
```diff
@@ -6,16 +6,7 @@ import { recordTracksEvent } from '@automattic/calypso-analytics';
...
-	useSubmitTicketMutation,
-	useSubmitForumsMutation,
-	useSiteAnalysis,
-	useUserSites,
-	AnalysisReport,
-	SiteDetails,
-	HelpCenterSite,
-	useJetpackSearchAIQuery,
-} from '@automattic/data-stores';
+import { SiteDetails, HelpCenterSite } from '@automattic/data-stores';
```
Comment (tyxla) [inline packages/help-center/src/components/help-center-contact-form.tsx]: These are both types and can be moved to the `import type` import statement that imports from `@automattic/data-stores` below.

File: packages/help-center/src/components/help-center-notice.tsx [hunk 4]
```diff
@@ -1,10 +1,10 @@
-import { SupportActivity } from '@automattic/data-stores';
 import { localizeUrl, useLocale, getRelativeTimeString } from '@automattic/i18n-utils';
 import { ExternalLink } from '@wordpress/components';
 import { createInterpolateElement } from '@wordpress/element';
 import { __, sprintf } from '@wordpress/i18n';
 import { Icon, info } from '@wordpress/icons';
-import type { AnalysisReport } from '@automattic/data-stores';
+import { SupportActivity } from '../data/types';
```
Comment (tyxla) [inline packages/help-center/src/components/help-center-notice.tsx]: I think we can just move this to the next line, since it's also a type and imports from the same module.

File: packages/help-center/src/data/use-submit-forums-topic.ts [hunk 5]
```diff
@@ -1,8 +1,8 @@
+import { SiteDetails } from '@automattic/data-stores';
```
Comment (tyxla) [inline packages/help-center/src/data/use-submit-forums-topic.ts]: Could be `import type`.

File: packages/help-center/src/data/types.ts [hunk 6]
```diff
@@ -1,3 +1,5 @@
+import type { SiteDetails, HelpCenterSite } from '@automattic/data-stores';
```
Comment (tyxla) [inline packages/help-center/src/data/types.ts]: I wonder if we could just merge `./data/types.ts` with `./types`. With the move it no longer makes sense to have 2 separate files for those types IMHO.

File: packages/data-stores/src/queries/use-site-intent.ts (hunk 2 above)
Comment (tyxla) [reply packages/data-stores/src/queries/use-site-intent.ts]: There are a few comments throughout the codebase that will no longer make sense and we might need to update them:

https://github.com/Automattic/wp-calypso/blob/a637ef7503821ee8bcaf6972e3dd41febca2c136/apps/editing-toolkit/editing-toolkit-plugin/dotcom-fse/lib/site-intent/use-site-intent.js#L1

https://github.com/Automattic/wp-calypso/blob/a637ef7503821ee8bcaf6972e3dd41febca2c136/apps/wpcom-block-editor/src/wpcom/features/use-site-intent.js#L1

File: packages/data-stores/src/index.ts [hunk 7]
```diff
@@ -12,18 +12,7 @@ import * as Subscriber from './subscriber';
...
-export { useJetpackSearchAIQuery } from './queries/use-jetpack-search-ai';
-export { useSiteAnalysis } from './queries/use-site-analysis';
-export { useUserSites } from './queries/use-user-sites';
-export type { AnalysisReport } from './queries/use-site-analysis';
-export { useSiteIntent } from './queries/use-site-intent';
-export { useSupportAvailability } from './support-queries/use-support-availability';
-export { useSubmitTicketMutation } from './support-queries/use-submit-support-ticket';
-export { useSubmitForumsMutation } from './support-queries/use-submit-forums-topic';
-export { useUpdateZendeskUserFieldsMutation } from './support-queries/use-update-zendesk-user-fields';
-export { useSupportActivity } from './support-queries/use-support-activity';
```
Comment (tyxla) [inline packages/data-stores/src/index.ts]: Nice 😍

File: packages/data-stores/CHANGELOG.md [hunk 8]
```diff
@@ -1,5 +1,21 @@
 # Changelog
 
+## Unreleased
+
+### Breaking changes
+
+- Remove all custom query and mutation hooks (based on `@tanstack/query`) from `support-queries/*`. These are moved to `@automattic/help-center`.
+- Move the following query and mutation hooks (based on `@tanstack/query`) from `queries/*` to `@automattic/help-center`
+  - `use-is-wporg-site.ts`
```
Comment (tyxla) [inline packages/data-stores/CHANGELOG.md]: Nit: maybe it could be more helpful to list the hook names and not the file names here.

**Review by tyxla — COMMENTED** [review]
Nice refactor 👍 I think moving the custom queries makes a lot of sense and I've always found it unexpected to see them live in the `@automattic/data-stores` package.

This mostly looks good, but I've left a few suggestions, let me know what you think.

File: packages/data-stores/src/queries/use-site-intent.ts (hunk 2 above)
Comment (flootr) [reply packages/data-stores/src/queries/use-site-intent.ts]: Mmh, it may be worthwhile exploring if that's still something we want to do. Looping in @okmttdhr & @arthur791004 as they discussed this in https://github.com/Automattic/wp-calypso/pull/73565#discussion_r1113839120 – what do you think? We removed the `useSiteIntent` from the `data-stores` package as it doesn't seem to be used anywhere. We still think the `data-stores` package isn't the right fit for the hook regardless.

Comment (flootr) [reply packages/data-stores/src/queries/use-site-intent.ts]: Seems like [usages](https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/plans-features-main/hooks/use-intent-from-site-meta.ts) of `useSitenIntent` were introduced after I opened this PR. I reverted the commit which removed it with https://github.com/Automattic/wp-calypso/pull/78682/commits/c43cb08749d1659c039e338ed48d8dec83bb1757 but we still should find a better place for it.

**Review by tyxla — APPROVED** [review]
This looks great 🚀 

It would be great if we can get a thumbs-up from @Automattic/vertex.

Also, a side note: some of those hooks might be quite generic and could fit better in a package whose sole purpose is to contain custom queries. That could help if someone would like to utilize them in the future, be it in Calypso's client, in a package/app or outside of Calypso entirely.
